main.cpp

```diff
@@ -1,207 +1,129 @@
-            isPlaying = true;
-
-            cout << "Audio file playing...\n\n";
-
-            // restart track
-        } else if (n == 4 && !isPlaying) {
-            // place track to start
-            mciSendString("seek mp3 to start", nullptr, 0, nullptr);
-
-            // save current volume
-            mciSendStringA(("setAudio mp3 volume to " + currentVolume).c_str(), nullptr, 0, nullptr);
-
-            mciSendString("play mp3", nullptr, 0, nullptr);
-
-            isPlaying = true;
-
-
-            // change volume
-        } else if (n == 5 && !isPlaying) {
-            currentVolume = changeVolume(currentVolume);
-
-            isPlaying = true;
-
-            // rewind track
-        } else if (n == 6) {
-
-            // save current volume
-            mciSendStringA(("setAudio mp3 volume to " + currentVolume).c_str(), nullptr, 0, nullptr);
-
-            rewindTrack();
-
-            isPlaying = true;
-        }
-
-
-        if (isPlaying) {
-            cout << "Press 0 to pause the file and press 2 to exit the file." << endl;
-            cout << "To change track, change volume or start it over press pause first" << endl;
-            cin >> n;
-
-            if (n == 0) {
-                //pause the audio file
-                isPlaying = false;
-                mciSendString("pause mp3", nullptr, 0, nullptr);
-
-            } else if (n == 2) {
-                //close the audio file
-                mciSendString("close mp3", nullptr, 0, nullptr);
-                break;
-            }
-
-        }
-    }
-    return 0;
-}
-
-
-void getFileNames(const string &folder, vector<string> *files) {
-    vector<string> names;
-    DIR *dir;
-    struct dirent *ent;
-
-    if ((dir = opendir(folder.c_str())) != nullptr) {
-        /* print all the files and directories within directory */
-        while ((ent = readdir(dir)) != nullptr) {
-            if (!(strcmp(ent->d_name, "..")) || !(strcmp(ent->d_name, "."))) {
-                (void) 0;
-            } else {
-                names.emplace_back(ent->d_name);
-            }
-        }
-        closedir(dir);
-        *files = names;
-    }
-}
-
-
-string changeTrack(vector<string> files) {
-    int counter, trackNumber;
-    string trackName, fileLink;
-
-    // output files
-    cout << "\nfiles in folder 'music':" << endl;
-    counter = 0;
-    for (auto const &file: files) {
-        counter++;
-        cout << counter << ". " << file << endl;
-    }
-    cout << "\nTo choose track press it's number." << endl;
-    cin >> trackNumber;
-
-    // check that trackNumber is valid
-    if (trackNumber > counter || isdigit(trackNumber) == trackNumber) {
-        cout << "Press correct track's number";
-    }
-
-    trackName = files[trackNumber - 1];
-    fileLink = "open music/" + trackName + " type mpegVideo alias mp3";
-
-    return fileLink;
+  AudioPlayer audioPlayer;
+  bool firstIteration = true;
+  string volume = "50";
```

Comment:
Define it where it should be used - here https://github.com/ANANAzZzZz/Music-Player/pull/2/files#diff-608d8de3fba954c50110b6d7386988f27295de845e9d7174e40095ba5efcf1bbR76.

Defining variables at the beginning of the function is bad

Reply:
✅